DashAI/back/dataloaders/classes/dataloader.py

```diff
@@ -0,0 +1,70 @@
+from abc import abstractmethod
+
+from datasets import ClassLabel
+
+from DashAI.back.config_object import ConfigObject
+
+
+class DataLoader(ConfigObject):
+    """
+    Abstract class with base methods for all data loaders
+    """
+
+    @abstractmethod
+    def load_data(self, dataset_path, url=None):
+        pass
+
+    @abstractmethod
+    def set_task_format(self, task):
+        pass
+
+    def split_dataset(self, dataset, params, class_column):
+        """
+        Split the dataset in train, test and validation data
+        """
+        stratify_column = None
+        if params["stratify"]:
+            stratify_column = class_column
+        test_val = params["test_size"] + params["val_size"]
+        val_size = params["val_size"] / test_val
```

Comment:
This is because the splitting are made twice. First the dataset is splitting into a train split and a test-validation split, this is because the validation split is taken like a portion of the test split of the first splitting. For that the size of the test split for the first splitting is the sum of the sizes of test and validation splits.

Then, the test and validation splits are defined in the second splitting, where now the train size of this splitting is the final test split and the result of the test split is actually the validation split. So that ‘val_size’ is the proportion of the validation split of the rest of the data in the resulting test split form first splitting process.

An example, if I have a  train size of 0.8, a test size of 0.1 and a validation size of 0.1, in the first process we split the data in the train data with the 80% of the data, and a test-validation data with the 20% remaining. Then in then second process we divide this 20% in a 10% of test data and 10% of validation data. 

This is all because Hugging Face datasets use a method called ‘train_test_split’ similar to sklearn method, and for do the validation split we need do this trick. As recommended in this [forum](https://discuss.huggingface.co/t/how-to-split-main-dataset-into-train-dev-test-as-datasetdict/1090).

A bad case for this, is when your have very few samples, cause some splits can be empty. But for this is checked in the validation of dataset format that do the task, that you have an appropriate number of samples.